consensus/polybft/state.go

```diff
@@ -0,0 +1,692 @@
+package polybft
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"sort"
+
+	"github.com/0xPolygon/pbft-consensus"
+	"github.com/boltdb/bolt"
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/abi"
+)
+
+/*
+The client has a boltDB backed state store. The schema as of looks as follows:
+
+events/
+|--> stateSyncEvent.Id -> *StateSyncEvent (json marshalled)
+
+commitments/
+|--> commitment.Message.ToIndex -> *CommitmentMessageSigned (json marshalled)
+
+bundles/
+|--> bundle.StateSyncs[0].Id -> *BundleProof (json marshalled)
+
+epochs/
+|--> epochNumber
+	|--> hash -> []*MessageSignatures (json marshalled)
+
+validatorSnapshots/
+|--> epochNumber -> *AccountSet (json marshalled)
+*/
+
+var (
+	// ABI
+	stateTransferEvent      = abi.MustNewEvent("event StateSync(uint256 indexed id, address indexed sender, address indexed target, bytes data)") //nolint:lll
+	onStateReceiveMethod, _ = abi.NewMethod("function onStateReceive(uint64, address, bytes)")
+)
+
+const (
+	// validatorSnapshotLimit defines a maximum number of validator snapshots
+	// that can be stored in cache (both memory and db)
+	validatorSnapshotLimit = 100
+	// numberOfSnapshotsToLeaveInMemory defines a number of validator snapshots to leave in memory
+	numberOfSnapshotsToLeaveInMemory = 2
+	// numberOfSnapshotsToLeaveInMemory defines a number of validator snapshots to leave in db
+	numberOfSnapshotsToLeaveInDB = 10
+	// number of stateSyncEvents to be processed before a commitment message can be created and gossiped
+	stateSyncMainBundleSize = 5
+	// number of stateSyncEvents to be grouped into one StateTransaction
+	stateSyncBundleSize = 5
+)
+
+// StateSyncEvent is a bridge event from the rootchain
+type StateSyncEvent struct {
+	// ID is the decoded 'index' field from the event
+	ID uint64
+	// Sender is the decoded 'sender' field from the event
+	Sender ethgo.Address
+	// Target is the decoded 'target' field from the event
+	Target ethgo.Address
+	// Data is the decoded 'data' field from the event
+	Data []byte
+	// Log contains raw data about smart contract event execution
+	Log *ethgo.Log
```

Comment:
Please use the default log type instead of `ethgo.Log`

Reply:
We rely on `Tracker` from the `ethgo` library for tracking events emitted on the rootchain and it provides us `ethgo.Log` objects (since it doesn't know for `types.Log` from the Edge codebase). Therefore unsure whether it is feasible to make such change (I mean we could, but we would need to have conversion from one to another type implemented somewhere and doesn't seem like beneficial, because we couldn't get rid of `etgo.Log` completely).

Reply:
Agree with Stefan.